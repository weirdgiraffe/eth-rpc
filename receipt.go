package ethrpc

import (
	"context"
	"encoding/json"

	"github.com/pkg/errors"
)

type TransactionReceipt struct {
	BlockHash   Hash   `json:"blockHash"`
	BlockNumber Number `json:"blockNumber"`

	TransactionHash  Hash    `json:"transactionHash"`
	TransactionIndex Number  `json:"transactionIndex"`
	Type             Number  `json:"type"`
	From             Address `json:"from"`
	To               Address `json:"to"`
	Status           Number  `json:"status"`
	Root             Hash    `json:"root"`

	ContractAddress Address `json:"contractAddress"`

	GasUsed           Number `json:"gasUsed"`
	CumulativeGasUsed Number `json:"cumulativeGasUsed"`
	EffectiveGasPrice Number `json:"effectiveGasPrice"`

	LogsBloom Data       `json:"logsBloom"`
	Logs      []LogEntry `json:"logs"`
}

func (c *Client) GetTransactionReceipt(ctx context.Context, txHash Hash) (*TransactionReceipt, error) {
	res, err := c.impl.Call(ctx, "eth_getTransactionReceipt", txHash)
	if err != nil {
		return nil, err
	}
	if res.Error != nil {
		return nil, res.Error
	}
	var out TransactionReceipt
	err = json.Unmarshal(res.Result, &out)
	if err != nil {
		return nil, errors.Wrap(err, "failed to decode rpc result")
	}
	return &out, nil
}
