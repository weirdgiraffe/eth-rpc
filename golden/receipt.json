{
    "blockHash": "0xb3b20624f8f0f86eb50dd04688409e5cea4bd02d700bf6e79e9384d47d6a5a35",
    "blockNumber": "0x5bad55",
    "contractAddress": null,
    "cumulativeGasUsed": "0xb90b0",
    "from": "0x398137383b3d25c92898c656696e41950e47316b",
    "gasUsed": "0x1383f",
    "logs": [
        {
            "address": "0x06012c8cf97bead5deae237070f9587f8e7a266d",
            "blockHash": "0xb3b20624f8f0f86eb50dd04688409e5cea4bd02d700bf6e79e9384d47d6a5a35",
            "blockNumber": "0x5bad55",
            "data": "0x000000000000000000000000398137383b3d25c92898c656696e41950e47316b00000000000000000000000000000000000000000000000000000000000cee6100000000000000000000000000000000000000000000000000000000000ac3e100000000000000000000000000000000000000000000000000000000005baf35",
            "logIndex": "0x6",
            "removed": false,
            "topics": [
                "0x241ea03ca20251805084d27d4440371c34a0b85ff108f6bb5611248f73818b80"
            ],
            "transactionHash": "0xbb3a336e3f823ec18197f1e13ee875700f08f03e2cab75f0d0b118dabb44cba0",
            "transactionIndex": "0x11"
        }
    ],
    "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000800000200000000000000000000000000000",
    "status": "0x1",
    "to": "0x06012c8cf97bead5deae237070f9587f8e7a266d",
    "transactionHash": "0xbb3a336e3f823ec18197f1e13ee875700f08f03e2cab75f0d0b118dabb44cba0",
    "transactionIndex": "0x11",
    "type": "0x0"
}
